### PR TITLE
Login & Authentication Improvements

### PR DESCRIPTION
**Pull request by becky-intelletive**

### CODE REVIEW

File: dash/backend/src/guards/auth.guard.ts [hunk 1]
```diff
@@ -5,16 +5,23 @@ import {MineLoggerService} from '../modules/shared/services/mine-logger.service'
...
+        {
+          label: 'User authenticated',
+          data: {params: {userId: this._loggedInUser.id}}
+        },
+        AuthGuard.name
+      );
       return true;
     } else {
       throw new UnauthorizedException('Invalid auth token');
-      return false;
```
Comment (becky-intelletive) [inline dash/backend/src/guards/auth.guard.ts:17]: unreachable line

File: dash/backend/src/guards/auth.guard.ts [hunk 2]
```diff
@@ -5,16 +5,23 @@ import {MineLoggerService} from '../modules/shared/services/mine-logger.service'
 @Injectable()
 export class AuthGuard implements CanActivate {
 
-  constructor(@Inject('LOGGED_IN_USER') private readonly _loggedInUser: UserProfileDto,
-              private readonly logger: MineLoggerService){}
+  constructor(
```
Comment (becky-intelletive) [inline dash/backend/src/guards/auth.guard.ts:8]: file is just reformatted

File: dash/backend/src/modules/auth/controllers/default.controller.ts [hunk 3]
```diff
@@ -55,6 +58,19 @@ export class DefaultController {
         return null;
     }
 
+    @Get('check-status')
```
Comment (becky-intelletive) [inline dash/backend/src/modules/auth/controllers/default.controller.ts:61]: new GET method that checks whether the user is logged in the same way the backend's AuthGuard does

File: dash/backend/src/modules/auth/controllers/non-redirectable-login.controller.ts [hunk 4]
```diff
@@ -28,18 +28,18 @@ import common from "../../../config/common";
...
 
-    constructor(private readonly externalAuthConfigService: ExternalAuthConfigService,
-                private readonly ldapFactory: LdapFactory,
-                private readonly userProfileService: UserProfileService,
-                private readonly jwtUtility: JwtUtilityService,
-                private readonly logger: MineLoggerService,
-                private readonly loginCounterService: LoginCounterService,
-                private readonly auditLogService: AuditLogService,
-    ){
+    private readonly CONTEXT = NonRedirectableLoginController.name;
```
Comment (becky-intelletive) [inline dash/backend/src/modules/auth/controllers/non-redirectable-login.controller.ts:32]: file is just reformatted

File: dash/backend/src/modules/auth/controllers/redirectable-login.controller.ts [hunk 5]
```diff
@@ -10,10 +10,12 @@ import { ConfigService } from '@nestjs/config';
 @Controller('redirectable-login')
 export class RedirectableLoginController {
 
-  constructor(private readonly oAuth2Factory: OAuth2Factory,
-              private readonly externalAuthConfigService: ExternalAuthConfigService,
-              private readonly jwtUtility: JwtUtilityService,
-              private readonly configService: ConfigService) {}
+  constructor(
```
Comment (becky-intelletive) [inline dash/backend/src/modules/auth/controllers/redirectable-login.controller.ts:13]: file is just reformatted

File: dash/backend/src/modules/auth/open-api-schema/authentication-meta-schema.ts [hunk 6]
```diff
@@ -72,6 +72,33 @@ export const AUTH_VALIDATE_RESPONSE_SCHEMA =  {
         }
     }
 };
+export const AUTH_CHECK_LOGIN_STATUS_RESPONSE_SCHEMA =  {
```
Comment (becky-intelletive) [inline dash/backend/src/modules/auth/open-api-schema/authentication-meta-schema.ts:75]: new schema type for the new auth status endpoint

File: dash/frontend/src/app/app.component.ts [hunk 7]
```diff
@@ -23,13 +23,14 @@ export class AppComponent implements OnInit, OnDestroy {
...
   @HostBinding('class') componentCssClass;
 
-  constructor(private themeService: ThemeService,
-              private overlayContainer: OverlayContainer,
-              private titleService: Title,
-              private jwtAuthService: JwtAuthService,
-              private router: Router,
-              private activatedRoute: ActivatedRoute) {
-  }
+  constructor(
```
Comment (becky-intelletive) [inline dash/frontend/src/app/app.component.ts:26]: this part is just reformatted

File: dash/frontend/src/app/app.component.ts [hunk 8]
```diff
@@ -43,13 +44,15 @@ export class AppComponent implements OnInit, OnDestroy {
...
     });
 
-    timer(0, 10000)
-      .pipe(takeUntil(this.unsubscribe$))
-      .subscribe(() => {
-        if (this.jwtAuthService.isTokenExpired()) {
-          this.router.navigate(['/login']);
-        }
-      });
+    // timer(0, 10000)
```
Comment (becky-intelletive) [inline dash/frontend/src/app/app.component.ts:47]: removed the timer that booted you out after a certain amount of time. can uncomment it if you want to keep it.

File: dash/frontend/src/app/core/route-guards/auth.guard.ts [hunk 9]
```diff
@@ -2,33 +2,58 @@ import { Injectable } from '@angular/core';
...
-  }
+  constructor(
+    private readonly router: Router,
+    private alertService: AlertService,
+    private httpClient: HttpClient,
+    private loaderService: NgxUiLoaderService,
+    private jwtAuthService: JwtAuthService,
+  ) {}
 
   canActivate(
```
Comment (becky-intelletive) [inline dash/frontend/src/app/core/route-guards/auth.guard.ts:24]: reworked this method to use the /auth/check-status endpoint. it no longer relies on a 401 to tell whether the user is logged out.

File: dash/frontend/src/app/core/route-guards/login.guard.ts [hunk 10]
```diff
@@ -2,30 +2,40 @@ import { Injectable } from '@angular/core';
...
-              private loaderService: NgxUiLoaderService) {
-  }
+  constructor(
+    private alertService: AlertService,
+    private httpClient: HttpClient,
+    private router: Router,
+    private jwtAuthService: JwtAuthService,
+  ) {}
 
   canActivate(
```
Comment (becky-intelletive) [inline dash/frontend/src/app/core/route-guards/login.guard.ts:22]: same as in AuthGuard

File: dash/frontend/src/app/core/services/jwt-auth.service.ts [hunk 11]
```diff
@@ -16,22 +16,34 @@ export class JwtAuthService {
...
       return decoded;
     } catch (e) {
       return null;
     }
   }
 
+  getCurrentUserData(): IUser {
+    try {
+      const rawToken = localStorage.getItem('__token__');
+      const decoded = this.parseToken(rawToken);
```
Comment (becky-intelletive) [inline dash/frontend/src/app/core/services/jwt-auth.service.ts:36]: uses existing method to parse the token

File: dash/frontend/src/app/modules/private/private-routing.module.ts [hunk 12]
```diff
@@ -62,7 +62,7 @@ const routes: Routes = [
     path: '',
```
Comment (becky-intelletive) [inline dash/frontend/src/app/modules/private/private-routing.module.ts:62]: file is just reformatted

File: dash/frontend/src/app/modules/public/pages/external-auth-cb/external-auth-cb.component.ts [hunk 13]
```diff
@@ -11,11 +11,13 @@ import { NgxUiLoaderService } from 'ngx-ui-loader';
 })
 export class ExternalAuthCbComponent implements OnInit, OnDestroy {
 
-  constructor(private route: ActivatedRoute,
-              private router: Router,
-              private authService: AuthService,
-              private jwtService: JwtAuthService,
-              private loaderService: NgxUiLoaderService) { }
+  constructor(
```
Comment (becky-intelletive) [inline dash/frontend/src/app/modules/public/pages/external-auth-cb/external-auth-cb.component.ts:14]: file is just reformatted

File: dash/frontend/src/app/modules/private/private-routing.module.ts [hunk 14]
```diff
@@ -313,275 +313,274 @@ const routes: Routes = [
...
-        canActivate: [RoleGuard],
-        data: {
-          title: 'Audit Logs',
-        }
+        },
+        children: [
+          {
+            path: '',
+            component: AuditLogComponent,
+            canActivate: [RoleGuard],
```
Comment (jacobbeasley) [inline dash/frontend/src/app/modules/private/private-routing.module.ts:574]: Did anything need to be updated in RoleGuard?

Comment (becky-intelletive) [reply dash/frontend/src/app/modules/private/private-routing.module.ts:574]: No. RoleGuard looks at the currently-logged-in user's permissions.

**Pull request merged**